pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -176,11 +181,18 @@ private TransformOperator buildTransformOperatorForFilteredAggregates(BaseFilter
     BaseFilterOperator filterOperator = filterPlanNode.run();
 
     // Use metadata/dictionary to solve the query if possible
-    // TODO: Use the same operator for both of them so that COUNT(*), MAX(col) can be optimized
```

Comment:
Don't remove this TODO because it is not solved yet (if a query contains half of the aggregation solvable by metadata, and another half solvable by dictionary, we cannot optimize it now). Since we are already doing the optimization, we can probably solve this TODO in this PR by combining both operators into one?

Reply:
Let me revisit this PR to unify these operators. I want to introduce a minimal interface which `Dictionary` and `DataSource` implement. In another PR, I completely remove the count optimisation here because it can be solved in a simpler way...

Reply:
I've merged these operators now, please take a look.